quarkus/src/main/java/com/redhat/openshift/knative/showcase/events/Endpoint.java

```diff
@@ -0,0 +1,52 @@
+package com.redhat.openshift.knative.showcase.events;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.jackson.JsonFormat;
+import io.smallrye.mutiny.Multi;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.jboss.resteasy.reactive.RestStreamElementType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.MediaType;
+
+@Path("events")
+@ApplicationScoped
+class Endpoint {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Endpoint.class);
+  private final EventStore events = new EventStore();
+
+  @GET
+  @Operation(summary = "Retrieves all registered events as a JSON stream")
+  @RestStreamElementType(JsonFormat.CONTENT_TYPE)
+  public Multi<byte[]> events() {
+    return events.stream()
+      .map(Endpoint::workaroundQuarkus31587);
+  }
+
+  @POST
+  @Consumes({MediaType.APPLICATION_JSON, JsonFormat.CONTENT_TYPE})
+  @Operation(summary = "Receives a CloudEvent and stores it")
+  public void receive(CloudEvent event) {
+    events.add(event);
+    LOGGER.debug("Received event: {}", event);
```

Comment:
We should log the event in the same way [the Knative's event-display image](https://knative.dev/docs/eventing/flows/sequence/sequence-reply-to-event-display/#inspecting-the-results) does it:

```
☁️  cloudevents.Event
Validation: valid
Context Attributes,
  specversion: 1.0
  type: samples.http.mode3
  source: /apis/v1/namespaces/default/pingsources/ping-source
  id: e8fa7906-ab62-4e61-9c13-a9406e2130a9
  time: 2020-03-02T20:52:00.0004957Z
  datacontenttype: application/json
Data,
  {
    "id": 0,
    "message": "Hello world"
  }
```

Reply:
From my point of view, this could be done in a follow up PR.

Reply:
This PR shows how that's could be addressed: https://github.com/cardil/openshift-knative-showcase/pull/1